rfq/order.go

```diff
@@ -53,12 +54,19 @@ func parseHtlcCustomRecords(customRecords map[uint64][]byte) (*rfqmsg.Htlc,
 // SerialisedScid is a serialised short channel id (SCID).
 type SerialisedScid = rfqmsg.SerialisedScid
 
+// SpecifierChecker checks whether the passed specifier and asset ID match. If
+// the specifier contains a group key, it will check whether the asset belongs
+// to that group.
+type SpecifierChecker func(ctx context.Context, specifier asset.Specifier,
+	id asset.ID) (bool, error)
+
 // Policy is an interface that abstracts the terms which determine whether an
 // asset sale/purchase channel HTLC is accepted or rejected.
 type Policy interface {
 	// CheckHtlcCompliance returns an error if the given HTLC intercept
 	// descriptor does not satisfy the subject policy.
-	CheckHtlcCompliance(htlc lndclient.InterceptedHtlc) error
+	CheckHtlcCompliance(htlc lndclient.InterceptedHtlc,
```

Comment:
I think since we're touching this method signature anyway, let's add the context here.